components/motor/esp_simplefoc/test_apps/main/test_esp_simplefoc.cpp

```diff
@@ -42,6 +42,21 @@ TEST_CASE("test mt6701", "[sensor][mt6701]")
     mt6701.deinit();
 }
 
+TEST_CASE("test as5048a", "[sensor][as5048a]")
+{
+#if CONFIG_IDF_TARGET_ESP32C3
+    AS5048a as5048a = AS5048a(SPI2_HOST, GPIO_NUM_2, GPIO_NUM_1, (gpio_num_t) -1, GPIO_NUM_3);
+#else
+    AS5048a as5048a = AS5048a(SPI2_HOST, GPIO_NUM_2, GPIO_NUM_1, (gpio_num_t) -1, GPIO_NUM_42);
```

Comment:
You need to add the header file in "components/motor/esp_simplefoc/port/esp/include/esp_simplefoc.h", otherwise it will cause compilation errors:
```shell
error: 'AS5048a' was not declared in this scope
   50 |     AS5048a as5048a = AS5048a(SPI2_HOST, GPIO_NUM_2, GPIO_NUM_1, (gpio_num_t) -1, GPIO_NUM_42);
```
You can refer to "esp_simplefoc.h":
```c
#include "esp_platform.h"
#include "BLDCMotor.h"
#include "esp_hal_bldc_3pwm.h"
#include "communication/SimpleFOCDebug.h"
#include "communication/Commander.h"
#include "sensors/GenericSensor.h"
#include "current_sense/GenericCurrentSense.h"
#include "current_sense/LowsideCurrentSense.h"
#include "../../angle_sensor/as5600.h"
#include "../../angle_sensor/mt6701.h"
#include "../../angle_sensor/as5048a.h"
```
In addition, you need to add your "as5048a.cpp" to "components/motor/esp_simplefoc/CMakeLists.txt", you can refer to:
```cmake
set(SRC_FILES
    "port/esp/esp_hal_misc.cpp"
    "port/esp/esp_hal_serial.cpp"
    "port/esp/esp_hal_gpio.cpp"
    "port/esp/esp_hal_bldc_3pwm.cpp"
    "port/esp/esp_app_fmath.cpp"
    "port/esp/esp_app_print.cpp"
    "port/angle_sensor/as5600.cpp"
    "port/angle_sensor/mt6701.cpp"
    "port/angle_sensor/as5048a.cpp")

set(INC_FILES "port/esp/include/" "port/angle_sensor")
set(IGNORE_FILES "port/esp/esp_hal_adc.cpp" "port/esp/esp_hal_mcpwm.cpp"
                 "port/esp/esp_hal_bldc_3pwm.cpp" "port/angle_sensor/mt6701.cpp"
                 "port/angle_snesor/as5048a.cpp")
```

Reply:
updated header & build files